### PR TITLE
virtctl vmexport: Improve garbage collection of vmexport resources

### PR DESCRIPTION
**Pull request by alromeros**
### What this PR does

This Pull Request improves the garbage collection of `virtualmachineexport` resources in virtctl so, by default, we only delete resources created during the command's lifecycle.

Previously, the decision to delete a `virtualmachineexport` was based on the provided flags without adequately accounting for pre-existing resources. This could lead to inconsistencies between the flags and the actual state of resources.

Fixes #

### Release note

```release-note
NONE
```

### CODE REVIEW

**Review by 0xFelix — COMMENTED** [review]
Can you add a unit test for this behavior?

**Review by 0xFelix — COMMENTED** [review]
One nit, otherwise looks good!

File: pkg/virtctl/vmexport/vmexport_test.go [hunk 1]
```diff
@@ -376,6 +377,43 @@ var _ = Describe("vmexport", func() {
 			Expect(err).ToNot(HaveOccurred())
 		})
 
+		DescribeTable("Succesfully create and download a VirtualMachineExport in different steps", func(expectedDeletes int, arg string) {
+			utils.HandleVMExportCreate(vmExportClient, nil)
+			cmd := clientcmd.NewRepeatableVirtctlCommand(commandName, virtctlvmexport.CREATE, vmexportName, setflag(virtctlvmexport.PVC_FLAG, "test-pvc"))
+			err := cmd()
+			Expect(err).ToNot(HaveOccurred())
```
[the remark this answers is not in the document]
Comment (alromeros) [reply pkg/virtctl/vmexport/vmexport_test.go]: ack

File: pkg/virtctl/vmexport/vmexport_test.go [hunk 2]
```diff
@@ -376,6 +377,43 @@ var _ = Describe("vmexport", func() {
...
+				return true, nil, nil
+			})
+
+			args := []string{commandName, virtctlvmexport.DOWNLOAD, vmexportName, setflag(virtctlvmexport.VOLUME_FLAG, volumeName), setflag(virtctlvmexport.OUTPUT_FLAG, "test-pvc"), virtctlvmexport.INSECURE_FLAG, "-n", metav1.NamespaceDefault}
+			if arg != "" {
+				args = append(args, arg)
+			}
+			cmd = clientcmd.NewRepeatableVirtctlCommand(args...)
+			err = cmd()
+			Expect(err).ToNot(HaveOccurred())
```
[the remark this answers is not in the document]
Comment (alromeros) [reply pkg/virtctl/vmexport/vmexport_test.go]: ack

**Review by 0xFelix — APPROVED** [review]
/lgtm

**Pull request merged**